source/platforms/BitBucketServer.ts

```diff
@@ -39,18 +39,32 @@ export class BitBucketServer implements Platform {
    * Fails the current build, if status setting succeeds
    * then return true.
    */
-  updateStatus = async (passed: boolean | "pending", message: string, url?: string): Promise<boolean> => {
+  updateStatus = async (
+    passed: boolean | "pending",
+    message: string,
+    url?: string,
+    dangerID?: string
+  ): Promise<boolean> => {
     const pr = await this.api.getPullRequestInfo()
     const { latestCommit } = pr.fromRef
+
     let state = passed ? "SUCCESSFUL" : "FAILED"
     if (passed === "pending") {
       state = "INPROGRESS"
     }
+
+    let name = "Danger"
+    if (process.env["PERIL_INTEGRATION_ID"]) {
+      name = "Peril"
+    } else if (dangerID) {
+      name = dangerID
+    }
+
     try {
       await this.api.postBuildStatus(latestCommit, {
         state: state,
         key: "danger.systems",
-        name: process.env["PERIL_INTEGRATION_ID"] ? "Peril" : "Danger",
+        name: name,
```

Comment:
```suggestion
        name,
```
I don’t remember the style guide here, but in my repo we omit the key when the value name is the same. Feel free to ignore this!